mcwb/types.py

```diff
@@ -40,14 +40,36 @@ class Vec3(NamedTuple):
     y: Number = 0
     z: Number = 0
 
-    def __add__(self, other: Vec3):
-        return type(self)(self.x + other.x, self.y + other.y, self.z + other.z)
+    def __add__(self, other):
+        if type(other) is Vec3:
+            return type(self)(self.x + other.x, self.y + other.y, self.z + other.z)
+        else:
+            return type(self)(self.x + other, self.y + other, self.z + other)
+
+    def __sub__(self, other):
+        if type(other) is Vec3:
+            return type(self)(self.x - other.x, self.y - other.y, self.z - other.z)
+        else:
+            return type(self)(self.x - other, self.y - other, self.z - other)
+
+    def __mul__(self, other):
+        if type(other) is Vec3:
+            return type(self)(self.x * other.x, self.y * other.y, self.z * other.z)
+        else:
+            return type(self)(self.x * other, self.y * other, self.z * other)
+
+    def __truediv__(self, other):
+        if type(other) is Vec3:
+            return type(self)(self.x / other.x, self.y / other.y, self.z / other.z)
+        else:
+            return type(self)(self.x / other, self.y / other, self.z / other)
+
+    def to_int(self):
```

Comment:
I am not too fond of the name `to_int` as it does not return an `int`, but a `Vec3` with ints.
Maybe `with_ints()` or `coerce_int()` would be better.
Also please add a docstring.